### PR TITLE
Add option to configure Presto and Starburst Presto server ports

### LINKED ISSUE #533: Dataproc with starburst-presto initialization is failing
Error while initializing with starburst-presto:

```
May 21 10:40:57 presto-hadoop-m systemd[1]: Started Presto DB.
+ wait_for_presto_cluster_ready
+ (( i = 0 ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:40:58 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:41:10 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:41:22 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:41:33 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:41:45 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:41:56 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
]
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[296], content-type=[text/html; charset=ISO-8859-1], date=[Tuesday, May 21, 2019 10:42:07 AM UTC], server=[Jetty(9.2.15.v20160210)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>
```

### PR DESCRIPTION
**Pull request by ChethanUK**
Fixes #533

### CODE REVIEW

**Comment by medb** [comment]
Cluster creation fails in presto tests with this change, please fix them:
```
Jul 10 22:53:16 test-presto-ha-1-1-20190710-225046-ld7u-m-0 systemd[1]: Starting Presto DB...
Jul 10 22:53:16 test-presto-ha-1-1-20190710-225046-ld7u-m-0 launcher.py[4153]: Started as 4156
Jul 10 22:53:16 test-presto-ha-1-1-20190710-225046-ld7u-m-0 systemd[1]: Started Presto DB.
+ wait_for_presto_cluster_ready
+ (( i = 0 ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:8080
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
```

**Comment by ChethanUK** [comment]
> please fix it:
Once PR [597](https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/pull/597/commits/c755d119fc95debb28e2818426173c78cab64b7e) is merged, this issue will be fixed.

**Comment by medb** [comment]
@ChethanUK changing port to 8060 will be breaking change and arguably it's less user-friendly because it will require specifying `--server=` property when running `presto` CLI.

May you modify this PR to make port configurable with GCE metadata instead?

**Comment by ChethanUK** [comment]
> May you modify this PR to make port configurable with GCE metadata instead?

Okay, we will once discuss, we previously agreed on 8060 because Cloud Dataproc Presto Component uses 8060 and Zepplin or other components also uses 8080

**Comment by medb** [comment]
@aniket486 WDYT about discussion above?

**Comment by ChethanUK** [comment]
> 8060 and Zepplin or other components also uses 8080

In the init script, query [presto '--execute=select * from system.runtime.nodes;'] will give the following error:
```
+ sleep 10
+ (( i++ ))
+ (( i < 12 ))
+ presto '--execute=select * from system.runtime.nodes;'
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={access-control-allow-credentials=[true], access-control-allow-headers=[authorization,Content-Type], access-control-allow-methods=[POST, GET, OPTIONS, PUT, HEAD, DELETE], cache-control=[must-revalidate,no-cache,no-store], content-length=[329], content-type=[text/html;charset=iso-8859-1], date=[Thursday, July 11, 2019 11:55:34 AM UTC], server=[Jetty(9.4.14.v20181114)], x-frame-options=[SAMEORIGIN], x-xss-protection=[1]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p><hr><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.4.14.v20181114</a><hr/>
```
Hence, query should be: `if presto --server="localhost:${HTTP_PORT}" --execute='select 1'; then`

**Comment by medb** [comment]
@ChethanUK Query already fixed in this PR:
```
presto --server="localhost:${HTTP_PORT}" --execute='select * from system.runtime.nodes;'
```

Tests are passing, so I assume that it good to merge, if @aniket486@ will approve it.

**Comment by ChethanUK** [comment]
> less user-friendly
But I think either in CLI, JDBC connection or Apache Superset (or Redash, etc..) we have to give full connection string with IP

When we want to use it outside the cluster, it won't matter even if we use 8080 also...
```
Chethans-MacBook-Pro:Downloads chethan$ ./presto --server=**.***.***.**5
presto> show tables;
Error running command: java.net.ConnectException: Failed to connect to /**.***.***.**5:80
```

When the port is specified properly it will work
```
Chethans-MacBook-Pro:Downloads chethan$ ./presto --server=**.***.***.**5:8080
presto> show catalogs;
   Catalog    
--------------
 jmx
 .
 . 
 .
 system       
(9 rows)

Query 20190711_172234_00012_c989x, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]          
```

File: presto/README.MD [hunk 1]
```diff
@@ -11,13 +11,14 @@ You can use this initialization action to create a new Dataproc cluster with Pre
...
       --initialization-actions gs://dataproc-initialization-actions/presto/presto.sh
     ```
-1. Once the cluster has been created, Presto is configured to run on port `8080` (though you can change this in the script) on the master node in a Cloud Dataproc cluster. To connect to the Presto web interface, you will need to create an SSH tunnel and use a SOCKS 5 Proxy as described in the [dataproc web interfaces](https://cloud.google.com/dataproc/cluster-web-interfaces) documentation. You can also use the [Presto command line interface](https://prestodb.io/docs/current/installation/cli.html) using the `presto` command on the master node.
+1. Once the cluster has been created, Presto is configured to run on port `8060` (though you can change this in the script) on the master node in a Cloud Dataproc cluster. To connect to the Presto web interface, you will need to create an SSH tunnel and use a SOCKS 5 Proxy as described in the [dataproc web interfaces](https://cloud.google.com/dataproc/cluster-web-interfaces) documentation. You can also use the [Presto command line interface](https://prestodb.io/docs/current/installation/cli.html) using the `presto` command on the master node.
```
Comment (findepi) [inline presto/README.MD]: Should this be reverted to 8080?

File: presto/presto.sh [hunk 2]
```diff
@@ -29,8 +29,8 @@ else
   readonly CONNECTOR_JAR="$(find /usr/lib/hadoop/lib -name 'gcs-connector-*.jar')"
 fi
 readonly PRESTO_BASE_URL=https://repo1.maven.org/maven2/com/facebook/presto
-readonly PRESTO_VERSION='0.206'
-readonly HTTP_PORT='8080'
+readonly PRESTO_VERSION='0.221'
```
Comment (findepi) [inline presto/presto.sh]: unrelated change?

File: presto/presto.sh [hunk 3]
```diff
@@ -29,8 +29,8 @@ else
...
+readonly HTTP_PORT="$(/usr/share/google/get_metadata_value attributes/dataproc-presto-port || 8080)"
```
Comment (findepi) [inline presto/presto.sh]: This can't work, but I don't know the proper way of using attributes. @medb ?

File: presto/test_presto.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def __create_schema_via_hive(self, name, schema):
 
     def __verify_schema_via_presto(self, name, schema):
         query = "show schemas;"
-        presto_cmd = "presto --catalog=hive --execute='{}' --output-format TSV".format(
+        presto_cmd = "presto --server=localhost:8060 --catalog=hive --execute='{}' --output-format TSV".format(
```
Comment (findepi) [inline presto/test_presto.py]: Likely, the tests need no change, since default is now back to Presto default.

File: starburst-presto/README.MD [hunk 5]
```diff
@@ -18,6 +18,6 @@ You can find more information about using initialization actions with Dataproc i
 ## Important notes
 * This script must be updated based on which Presto version you wish to install
 * You may need to adjust the memory settings in `jvm.config` based on your needs
-* Presto is set to use HTTP port `8080` by default
+* Presto is set to use HTTP port `8080` by default, but can be changed using `--metadata dataproc-presto-port=8060`
```
Comment (findepi) [inline starburst-presto/README.MD]: @medb what's the convention here? Should this be prefixed with `dataproc-`? `--metadata presto-port=8060` could be sufficient to me.

File: starburst-presto/presto.sh [hunk 6]
```diff
@@ -23,7 +23,7 @@ readonly PRESTO_MASTER_FQDN="$(/usr/share/google/get_metadata_value attributes/d
 readonly WORKER_COUNT=$(/usr/share/google/get_metadata_value attributes/dataproc-worker-count)
 readonly PRESTO_MAJOR_VERSION="312"
 readonly STARBURST_PRESTO_VERSION="312-e.1"
-readonly HTTP_PORT="8080"
+readonly HTTP_PORT="$(/usr/share/google/get_metadata_value attributes/dataproc-presto-port || 8080)"
```
Comment (findepi) [inline starburst-presto/presto.sh]: Same comment as in `presto/presto.sh`

File: presto/presto.sh (hunk 3 above)
Comment (medb) [reply presto/presto.sh]: We need to replace `8080` with `echo 8080` to make it work

File: presto/test_presto.py (hunk 4 above)
Comment (medb) [reply presto/test_presto.py]: Yes, for init actions we don't need to have `dataproc` prefix.

File: presto/presto.sh (hunk 3 above)
Comment (findepi) [reply presto/presto.sh]: @medb is `/usr/share/google/get_metadata_value attributes/dataproc-presto-port` an executable or a plain text file?
If the latter, we need:
```
...=$(cat /usr/share/google/get_metadata_value attributes/dataproc-presto-port || echo 8080)
```

BTW Does Dataproc come woth some utility command for getting init param value (or default) in some easier way?

Comment (medb) [reply presto/presto.sh]: `/usr/share/google/get_metadata_value` is an executable and `attributes/dataproc-presto-port` its parameter - this is client for [GCE metadata service](https://cloud.google.com/compute/docs/storing-retrieving-metadata).

No, we intentionally do not provide utility bash functions to not create backward-compatibility problems, we usually define such functions in each init action when necessary.

Comment (findepi) [reply presto/presto.sh]: sorry, i misread that as/a/very/long/path/to/a/file

File: starburst-presto/README.MD (hunk 5 above)
Comment (medb) [reply starburst-presto/README.MD]: Done.
No need for `dataproc-` prefix.

File: presto/test_presto.py (hunk 4 above)
Comment (medb) [reply presto/test_presto.py]: Done.

File: starburst-presto/presto.sh (hunk 6 above)
Comment (medb) [reply starburst-presto/presto.sh]: Done.

File: presto/presto.sh (hunk 3 above)
Comment (medb) [reply presto/presto.sh]: Ack.

File: presto/README.MD (hunk 1 above)
Comment (medb) [reply presto/README.MD]: Done.

File: presto/presto.sh (hunk 2 above)
Comment (medb) [reply presto/presto.sh]: Reverted.

**Comment by mattsfuller** [comment]
Thanks!
